slang-com-ptr.h

```diff
@@ -111,7 +111,9 @@ class ComPtr
 protected:
 	/// Gets the address of the dumb pointer.
     // Disabled: use writeRef and readRef to get a reference based on usage.
```

Comment:
This was disabled because doing & could then overwrite the contained pointer with a new value without altering the ref. Is that not possible with this code generation path?

Reply:
You are right. In this case I really need a `ComPtr<T>*` from the & operator so it can be passed to an `out` parameter. Perhaps I can just do `(ComPtr<T>*)(ptr.writeRef())`

Reply:
Actually, the code generation does the following:
```
void createFoo(out IFoo f) { f = ...; }
void main()
{
     IFoo f;
     createFoo(f);
}
```

Lowers to 
```
void createFoo(ComPtr<IFoo> *f)
{
   *f = ...;
}
void main()
{
     ComPtr<IFoo> f;
     createFoo(&f);
}
```

Which should be fine regarding to write refs. Because without that `=delete` line, operator & returns a `ComPtr<T>*` instead of `T**` so it should be fool proof against mishandling ref counts.